### PR TITLE
fix: malformed YAML results in empty config set

### PR DESCRIPTION
**Pull request by GGabriele**
https://github.com/Kong/deck/issues/578

### CODE REVIEW

File: file/testdata/badyamlwithspace/bar.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+ _info:
+  select_tags:
```
[the remark this answers is not in the document]
Comment (GGabriele) [reply file/testdata/badyamlwithspace/bar.yml:2]: No idea, if you put the content of this file in two different online validators, you get two different results :P 
- https://codebeautify.org/yaml-validator
- http://www.yamllint.com/

[the remark this answers is not in the document]
Comment (rainest) [reply file/testdata/badyamlwithspace/bar.yml:2]: https://github.com/go-yaml/yaml/issues?page=2&q=is%3Aissue+space doesn't have anything that obviously looks like this issue.

My intuition is that it is not valid and should result in a parse error, but I am not an expert on YAML grammar and didn't try to exhaustively prove it one way or the other. A brief reading of https://yaml.org/spec/1.2.2/#chapter-6-structural-productions suggests that the only valid indented lines that aren't properly nested are comments. I expect it's quite unlikely that the YAML spec has a "if there are nested objects without a parent, this is valid YAML, but you must discard all other objects" gotcha in the spec.

IMO we just do this with an additional comment that it's to work around an as-yet unknown issue in go-yaml or ghodss/yaml. If we want to chase it further we could try to investigate parsing in those and submit an issue or bug upstream.

**Review by rainest — DISMISSED** [review]

File: file/testdata/badyamlwithspace/bar.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (GGabriele) [reply file/testdata/badyamlwithspace/bar.yml:2]: Added a comment here https://github.com/Kong/deck/pull/590/commits/0777131192351c4c4cb4e1d1773d46019dee3ace

File: file/readfile.go [hunk 2]
```diff
@@ -95,6 +102,11 @@ func readContent(reader io.Reader) (*Content, error) {
 	if err != nil {
 		return nil, fmt.Errorf("parsing file: %w", err)
 	}
+	// go-yaml implementation fails at correctly parsing a file whose first
+	// character is a space. If that is the case here, raise an error.
```
Comment (rainest) [inline file/readfile.go:106]:
```suggestion
	// character is a space, as shown in https://github.com/Kong/deck/issues/578
	// If that is the case here, raise an error.
```

**Review by rainest — CHANGES_REQUESTED** [review]

**Review by rainest — APPROVED** [review]

**Pull request merged**